src/utils/parser.cpp

```diff
@@ -0,0 +1,120 @@
+        std::cout << "--Playing Surface: " << team[7] << std::endl;
+        std::cout << "-----------League: " << team[8] << std::endl;
+        std::cout << "------Date opened: " << team[9] << std::endl;
+        std::cout << "--Distance2CtrFld: " << team[10] << std::endl;
+        std::cout << "Ballpark typology: " << team[11] << std::endl;
+        std::cout << "--------Roof Type: " << team[12] << std::endl;
+        std::cout << "NumberOSouvenirs:  " << team[13] << std::endl;
+        std::cout << "Souvenir ID:  " << team[14] << std::endl;
+        std::cout << "Is deleted?:  " << team[15] << std::endl;
+        std::cout << "---Souvenir:  " << team[16] << std::endl;
+        std::cout << "------Price:  " << team[17] << std::endl;
+        std::cout << "Souvenir ID:  " << team[18] << std::endl;
+        std::cout << "Is deleted?:  " << team[19] << std::endl;
+        std::cout << "---Souvenir:  " << team[20] << std::endl;
+        std::cout << "------Price:  " << team[21] << std::endl;
+        std::cout << "Souvenir ID:  " << team[22] << std::endl;
+        std::cout << "Is deleted?:  " << team[23] << std::endl;
+        std::cout << "---Souvenir:  " << team[24] << std::endl;
+        std::cout << "------Price:  " << team[25] << std::endl;
+        std::cout << "Souvenir ID:  " << team[26] << std::endl;
+        std::cout << "Is deleted?:  " << team[27] << std::endl;
+        std::cout << "---Souvenir:  " << team[28] << std::endl;
+        std::cout << "------Price:  " << team[29] << std::endl;
+        std::cout << "Souvenir ID:  " << team[30] << std::endl;
+        std::cout << "Is deleted?:  " << team[31] << std::endl;
+        std::cout << "---Souvenir:  " << team[32] << std::endl;
+        std::cout << "------Price:  " << team[33] << std::endl;
+        std::cout << std::endl;
+    }
+}
+
+void distanceParseDebug(const std::string& path)
+{
+    std::vector<std::vector<std::string>> distanceData = loadData(path);
+    for (const std::vector<std::string>& stadiumDistance : distanceData)
+    {
+
+        std::cout << "Stadium ID:   " << stadiumDistance[0] << std::endl;
+        std::cout << "Stadium Name: " << stadiumDistance[1] << std::endl;
+        std::cout << "Stadium ID:   " << stadiumDistance[2] << std::endl;
+        std::cout << "Stadium Name: " << stadiumDistance[3] << std::endl;
+        std::cout << "Distance:     " << stadiumDistance[4] << std::endl;
+        std::cout << std::endl;
+    }
+}
+
+std::vector<std::vector<std::string>> loadData(const std::string& path)
+{
+    // Initializing - vector will hold all lines of data within the document
+    std::vector<std::vector<std::string>> allRows;
+    std::ifstream infile(path);
+
+    if(path.substr((path.size()-4)) != ".csv"){throw BadFileFormat(QString::fromUtf8("Wrong File type\n"), QString::fromStdString(path));}
+    // Checks if file is open if not throw error
+    if (!infile.is_open()){throw BadFile(QFile(QString::fromUtf8("Wrong File Path\n")));}
```

Comment:
`BadFile(QFile)` takes a file path, not a message
```suggestion
    if (!infile.is_open()){throw BadFile(QFile(QString::fromStdString(path)));}
```